sdk/search/Azure.Search.Documents/README.md

```diff
@@ -96,6 +96,33 @@ string key = Environment.GetEnvironmentVariable("SEARCH_API_KEY");
 AzureKeyCredential credential = new AzureKeyCredential(key);
 SearchClient client = new SearchClient(endpoint, indexName, credential);
 ```
+### ASP.NET Core
+To inject `SearchClient` as a dependency in an ASP.NET Core app, first install the package `Microsoft.Extensions.Azure`. Then register the client in the `Startup.ConfigureServices` method:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddAzureClients(builder =>
+    {
+        builder.AddSearchClient(Configuration.GetSection("Search"));
+    });
+  
+    services.AddControllers();
+}
+```
+To use the preceding code, add this to your configuration:
+
+```json
+{
+    "Search": {
+      "endpoint": "https://<resource-name>.search.windows.net",
+      "credential": { "key": "resource key" },
```

Comment:
Is there a way to achieve this without hardcoding the API key in a config file? Or is "resource key" the name of an environment variable...?

Reply:
It's not the name of the environment variable. It is the key of the resource. @pakrym is the way to do what Bruce is asking?

Reply:
The IConfiguration system uses many sources of configuration, in ASP.NET Core one of them is environment variables. You can define the same key as SEARCH__CREDENTIAL__KEY="..." environment variable (the __ is used as a section separator).

Reply:
Thanks for the context. Can we clarify the example to make it clear that it follows that pattern? We don't want to encourage customers to hard-code secrets in config.

Reply:
I wonder if this feedback should apply to the other libraries as well, i.e. should we apply this to the Service Bus readme also?

Reply:
@joshfree I'm adding this to all the libraries that have the extension method for DI (some libraries don't from my investigation. I'll create issues for them.) I've added the section for SB already :)

Reply:
But the Service Bus readme uses the connection string in the config. In this Readme, it sounds like we are going to be updating it to not include secrets in the config. That is what I am wondering about.